### PR TITLE
Document a use case for Terraform

### PR DESCRIPTION
**Pull request by tpears**
# Contributor Comments

This branch introduces M2R2 and Sphinx which allow for the existing README to be ported to RST and for HTML to be auto-generated from the new RST documentation. The documentation site can be built locally after installing the updated pipenv dev packages by running the following command from the base directory:
`pipenv run sphinx-build -M html docs/ docs/_build`

You can then view the page by pointing a browser to your local file system. 

This branch was also built on readthedocs.io and can be viewed here: https://easy-infra.readthedocs.io/en/use-cases/

## Pull Request Checklist

Thank you for submitting a contribution to `easy_infra`!

In order to streamline the review of your contribution we ask that you review and comply with the below requirements:

- [x] Rebase your branch against the latest commit of the target branch, which likely should be `main`
- [x] If there is an issue associated with your Pull Request, align your PR title with [this documentation](https://help.github.com/en/articles/closing-issues-using-keywords)

### CODE REVIEW

File: docs/Terraform/index.rst [hunk 1]
```diff
@@ -0,0 +1,29 @@
+Terraform
+=========
+HashiCorp's Terraform is a tool for defining your Infrastructure as Code (IaC). The easy_infra project includes Terraform
+to support its use in CI/CD pipelines. You can use easy_infra to execute Terraform commands on an up-to-date version
```
Comment (JonZeolla) [inline docs/Terraform/index.rst:4]:
```suggestion
to support its use, both locally and in CI/CD pipelines. You can use easy_infra to execute Terraform commands, and
```

File: docs/Terraform/index.rst [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+of the program. As part of the terraform execution, security checks available from `Checkov <https://www.checkov.io/>`_, `tfsec <https://tfsec.dev/>`_, and `Terrascan <https://www.accurics.com/products/terrascan/>`_ are performed to assess your terraform code.
```
Comment (JonZeolla) [inline docs/Terraform/index.rst:5]:
```suggestion
as part of the execution, security checks available from `Checkov <https://www.checkov.io/>`_, `tfsec <https://tfsec.dev/>`_, and `Terrascan <https://www.accurics.com/products/terrascan/>`_ are transparently performed to assess your terraform code.
```

File: docs/Terraform/index.rst [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+Use Cases
+=========
+If you use Software Version Control to manage your Terraform IaC, consider executing ``terraform validate`` with 
+easy_infra as a pipeline action on commit or pull request::
+
+    docker run -v $(pwd):/tf seiso/easy_infra:latest-minimal terraform validate
```
Comment (JonZeolla) [inline docs/Terraform/index.rst:12]:
```suggestion
    docker run -v $(pwd):/iac seiso/easy_infra:latest-minimal terraform validate
```

File: docs/Terraform/index.rst [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+You can also use easy_infra to deploy your infrastructure using ``terraform plan`` and ``terraform deploy``::
+
+    docker run -v $(pwd):/tf seiso/easy_infra:latest-minimal /bin/bash -c "terraform plan && terraform apply -auto-approve"
```
Comment (JonZeolla) [inline docs/Terraform/index.rst:16]:
```suggestion
    docker run -v $(pwd):/iac seiso/easy_infra:latest-minimal /bin/bash -c "terraform plan && terraform apply -auto-approve"
```

File: docs/Terraform/index.rst [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+Terraform Cacheing
+------------------
+If you're working with the same terraform code across multiple runs, you can leverage the cache::
+
+    docker run -v $(pwd):/tf -v $(pwd)/plugin-cache:/root/.terraform.d/plugin-cache easy_infra:latest-minimal /bin/bash -c "terraform init; terraform version"
```
Comment (JonZeolla) [inline docs/Terraform/index.rst:22]:
```suggestion
    docker run -v $(pwd):/iac -v $(pwd)/plugin-cache:/root/.terraform.d/plugin-cache easy_infra:latest-minimal /bin/bash -c "terraform init; terraform validate"
```

File: docs/Terraform/index.rst [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+Resources
+=========
+Open Policy Agent (OPA) offers useful resources for cloud native infrastructure administrators.
```
Comment (JonZeolla) [inline docs/Terraform/index.rst:26]:
```suggestion
Configuring custom checks can be done by leveraging the robust Rego language, maintained by the
Open Policy Agent (OPA) project, and used as a part of `terrascan` for managing cloud native infrastructure.
```

File: docs/index.rst [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+.. mdinclude:: ../README.md
+
+.. toctree::
+   :maxdepth: 2
+   :caption: Contents:
+
+   Home <self>
+   Docker Images <Images/index>
+   Terraform/index
+   AWS Cli <awscli/index>
```
Comment (JonZeolla) [inline docs/index.rst:16]:
```suggestion
   AWS CLI <awscli/index>
```

File: docs/index.rst [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+   Azure CLI <az/index>
+   Packer/index
+   Ansible/index
+   Testing/index
+
+Indices and tables
+==================
+
+* :ref:`genindex`
+* :ref:`modindex`
```
Comment (JonZeolla) [inline docs/index.rst:26]: Broken link?

File: docs/index.rst [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+   AWS Cli <awscli/index>
+   Azure CLI <az/index>
+   Packer/index
+   Ansible/index
+   Testing/index
+
+Indices and tables
+==================
+
+* :ref:`genindex`
```
Comment (JonZeolla) [inline docs/index.rst:25]: Empty page?

File: docs/Terraform/index.rst [hunk 10]
```diff
@@ -0,0 +1,29 @@
+Terraform
+=========
+HashiCorp's Terraform is a tool for defining your Infrastructure as Code (IaC). The easy_infra project includes Terraform
+to support its use in CI/CD pipelines. You can use easy_infra to execute Terraform commands on an up-to-date version
+of the program. As part of the terraform execution, security checks available from `Checkov <https://www.checkov.io/>`_, `tfsec <https://tfsec.dev/>`_, and `Terrascan <https://www.accurics.com/products/terrascan/>`_ are performed to assess your terraform code.
+
+Use Cases
+=========
+If you use Software Version Control to manage your Terraform IaC, consider executing ``terraform validate`` with
```
Comment (JonZeolla) [inline docs/Terraform/index.rst:9]:
```suggestion
If you use Software Version Control (such as `git`) to manage your Terraform IaC, consider executing ``terraform validate`` with 
```

**Review by JonZeolla — CHANGES_REQUESTED** [review]
Looking pretty good, just some tweaks

File: docs/requirements.txt [hunk 11]
```diff
@@ -0,0 +1,5 @@
+#
+# This file defines requirements used by readthedocs.io
+#
+
+m2r2==0.2.7
```
[the remark this answers is not in the document]
Comment (JonZeolla) [reply docs/requirements.txt:5]: The way I've done this in the past is [this](https://github.com/SeisoLLC/easy_sast/blob/d99154ab42b91a8cdf22a2889913324ec007acc7/Makefile#L83).  I think we have some private projects too that do this via `tasks.py` already but I couldn't quickly find them

[the remark this answers is not in the document]
Comment (tpears) [reply docs/requirements.txt:5]: Tracking in https://github.com/SeisoLLC/easy_infra/issues/81

**Comment by tpears** [comment]
Missed comments that were collapsed. Accepted all the remaining suggestions.

File: docs/index.rst [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+.. mdinclude:: ../README.md
+
+.. toctree::
+   :maxdepth: 2
+   :caption: Contents:
+
+   Home <self>
+   Docker Images <Images/index>
+   Terraform/index
+   AWS ClI <awscli/index>
```
Comment (JonZeolla) [inline docs/index.rst:16]:
```suggestion
   AWS CLI <awscli/index>
```

**Review by JonZeolla — CHANGES_REQUESTED** [review]
@tpears please address the one typo comment I left and ensure that the details to deploy to read the docs are at least documented (I couldn't find it) if not automated as a part of CI/CD.  If needed we can do the automation as a follow-on, but need to ensure the docs render and deploy process is documented.

There's also a merge conflict to be addressed

**Comment by tpears** [comment]
Documentation for automation is in a follow task. Typo corrected and pipfile merged to rebuild pipfile.lock in preparation for merge.

**Review by JonZeolla — APPROVED** [review]
Looks great, nice work @tpears

**Pull request merged**